### ISSUE TITLE
BUG: gatsby-plugin-google-analytics is getting an undefined pathname

### ISSUE THREAD

**Issue by camsjams** [gatsbyjs/gatsby#1317]
Currently the [code](https://github.com/gatsbyjs/gatsby/blob/1.0/packages/gatsby-plugin-google-analytics/src/gatsby-browser.js) does this:

```javascript
exports.onRouteUpdate = function(location) {
  // Don't track while developing.
  if (process.env.NODE_ENV === `production`) {
    ga(`set`, `page`, location.pathname)
    ga(`send`, `pageview`)
  }
}
```

But it appears as though the parameter called `location` is structured like so:
```javascript
{
	location: {
		pathname: '/home',
		search: '',
		hash: '',
		key: 'q8junx'
	},
	action: 'PUSH'
}
```

The resulting bug is that we get production code doing something like:
```javascript
 ga('set', 'page', undefined);
```
Causing the GA metrics to show whatever the first SSR-served landing page to be the page viewed many times over (because `page` will never change).

Pretty easy change, but I am assuming this must have worked before?

Versions:
node v8.0.0
gatsby@1.0.0-beta.5
gatsby-plugin-google-analytics@1.0.0-beta.1

**Comment by KyleAMathews** [comment]
Oh wow... yeah the plugin is really outdated :-( the API got updated a while ago https://www.gatsbyjs.org/docs/browser-apis/#onRouteUpdate

**Comment by KyleAMathews** [comment]
could you submit a quick PR for this? I'm making a new release this afternoon so could get this out then.

**Comment by camsjams** [comment]
Sure thing, though I have found a secondary issue as well, you will see in PR.

**Comment by KyleAMathews** [comment]
Fixed in #1318
